### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by MishaLehotskyi**

### CODE REVIEW

File: src/transformState.js [hunk 1]
```diff
@@ -5,7 +5,31 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  const ADD_PROPERTIES = 'addProperties';
+  const REMOVE_PROPERTIES = 'removeProperties';
+  const CLEAR_PROPERTIES = 'clear';
```
Comment (andriy-shymkiv) [inline src/transformState.js]: why you hard coded all this type's instead of using switch case?

Comment (andriy-shymkiv) [reply src/transformState.js]: [checklist](https://github.com/mate-academy/js_stateful-object/blob/master/checklist.md) rule 2

File: src/transformState.js [hunk 2]
```diff
@@ -5,7 +5,31 @@
...
+
+  for (const action of actions) {
+    if (action.type === ADD_PROPERTIES) {
+      Object.assign(state, action.extraData);
+    }
+
+    if (action.type === REMOVE_PROPERTIES) {
+      for (const remove of action.keysToRemove) {
```
Comment (andriy-shymkiv) [inline src/transformState.js]:
```suggestion
      for (const key of action.keysToRemove) {
```

**Review by andriy-shymkiv — CHANGES_REQUESTED** [review]

**Review by xiimiik — APPROVED** [review]
Nice work!

**Pull request closed without merging**